File: mobile/tests/XPlatTests/YSMapTests.swift
//
//  YSMapTests.swift
//  XMailTests
//
//  Created by Aleksandr A. Dvornikov on 37//19.
//

import XCTest
@testable import XPlat

class YSMapTests: XCTestCase {
  let dictionary = YSMap(items: [1: "1", 2: "2", 3: "3"])

  func testShouldCorrectlyIterateMap() {
    var count: Int32 = 0
    dictionary.__forEach { value, key in
      XCTAssertEqual(value, dictionary.get(key))
      count += 1
    }
    XCTAssertEqual(count, dictionary.size)
  }
}

File: mobile/test/dependencies/YSMapTests.kt
package com.yandex.xplat.xmail

import com.yandex.xplat.common.*
import org.junit.Assert.*
import org.junit.Test

class YSMapTests {
    private val dictionary: YSMap<Int, String> = mutableMapOf(1 to "1", 2 to "2", 3 to "3")

    @Test
    fun shouldCorrectlyIterateMap() {
        var count = 0
        dictionary.__forEach { value, key ->
            assertEquals(value, dictionary.get(key))
            count++
        }
        assertEquals(dictionary.size, count)
    }
}
